apps/web/src/components/CookieBanner/CookieBanner.tsx

```diff
@@ -0,0 +1,142 @@
+import { getCookie, setCookie } from 'cookies-next'
+import Link from 'next/link'
+import { useRouter } from 'next/router'
+import type { MouseEventHandler } from 'react'
+import { useCallback, useEffect, useRef, useState } from 'react'
+
+import { SE_GDPR_COOKIE_ACCEPT_VALUE, SE_GDPR_COOKIE_NAME, SE_GDPR_COOKIE_REJECT_VALUE } from '@/constants/cookies'
+import { getGDPRCookieExpiryDate } from '@/utils/date'
+
+enum CookieBannerView {
+  Hidden,
+  Selection,
+  Accepted,
+  Rejected,
+}
+
+export function CookieBanner() {
+  const [view, setView] = useState(CookieBannerView.Hidden)
+
+  const regionRef = useRef<HTMLDivElement>(null)
+
+  const router = useRouter()
+
+  const handleRouteChange = useCallback(() => {
+    // Hide banner after navigating away from cookie policy page
+    const isCookieSet = Boolean(getCookie(SE_GDPR_COOKIE_NAME))
+
+    if (view === CookieBannerView.Selection && isCookieSet) {
+      setView(CookieBannerView.Hidden)
+    }
+    // Set initial visibility on client
+    if (view === CookieBannerView.Hidden && !isCookieSet) {
+      setView(CookieBannerView.Selection)
+    }
+
+    // Set focus on confirmation screen
+    if (view === CookieBannerView.Accepted || view === CookieBannerView.Rejected) {
+      regionRef.current?.focus()
+    }
+  }, [router, view])
+
+  useEffect(() => {
+    handleRouteChange()
+    router.events.on('routeChangeComplete', handleRouteChange)
+    return () => {
+      router.events.off('routeChangeComplete', handleRouteChange)
+    }
+  }, [handleRouteChange])
+
+  const handleAccept: MouseEventHandler = () => {
+    setView(CookieBannerView.Accepted)
+    handleDecision('granted', SE_GDPR_COOKIE_ACCEPT_VALUE)
+  }
+
+  const handleReject: MouseEventHandler = () => {
+    setView(CookieBannerView.Rejected)
+    handleDecision('denied', SE_GDPR_COOKIE_REJECT_VALUE)
+  }
+
+  const handleDecision = (cookieConsent: string, cookieValue: string) => {
+    setCookie(SE_GDPR_COOKIE_NAME, cookieValue, { expires: getGDPRCookieExpiryDate(), secure: true })
+    window.gtag('consent', 'update', {
+      ad_storage: cookieConsent,
+      analytics_storage: cookieConsent,
+      ad_user_data: cookieConsent,
+      ad_personalization: cookieConsent,
+    })
+  }
+
+  const handleHide: MouseEventHandler = () => {
+    setView(CookieBannerView.Hidden)
+  }
+
+  const renderSelection = () => {
+    return (
```

Comment:
if there is no logic then there is no need for the `{` followed by `return (`

`const renderSelection = () => (`

followed by jsx does the same job